### PR TITLE
add unit test for function IsEnvVarName

### PR DESCRIPTION
**Pull request by hanxiaoshuai**
**What this PR does / why we need it**:
add unit test for function IsEnvVarName
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/apimachinery/pkg/util/validation/validation_test.go [hunk 1]
```diff
@@ -511,3 +512,43 @@ func TestIsFullyQualifiedName(t *testing.T) {
...
+		"..",
+		"..EnvName",
+		"nospecialchars%^=@",
+		"env/name/slash",
+		"env.name/abc",
+		"env_name/abc",
+		"example.com/",
+		"/envname",
+		"1-starts-with-num",
+		"",
```
Comment (deads2k) [inline staging/src/k8s.io/apimachinery/pkg/util/validation/validation_test.go]: Add cases for colon and semicolon specifically please.  Those aren't allowed in bash shell per https://tiswww.case.edu/php/chet/bash/bash.html `name
A word consisting only of alphanumeric characters and underscores, and beginning with an alphabetic character or an underscore. Also referred to as an identifier.`

Probably worth linking to the man page too.

File: staging/src/k8s.io/apimachinery/pkg/util/validation/validation_test.go [hunk 2]
```diff
@@ -20,6 +20,7 @@ import (
 	"strings"
 	"testing"
 
+	"fmt"
```
Comment (dixudx) [inline staging/src/k8s.io/apimachinery/pkg/util/validation/validation_test.go]: Move this line to L20.

**Comment by dixudx** [comment]
@hanxiaoshuai Please squash your commits. Thanks.

**Comment by hanxiaoshuai** [comment]
@dixudx I have squashed.Thanks

**Comment by hanxiaoshuai** [comment]
cc @deads2k

**Comment by hanxiaoshuai** [comment]
cc @deads2k

**Pull request closed without merging**